ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,224 @@
+            var bugzillaRoot = "https://bugzilla-dev.allizom.org/"; // (prod is "https://bugzilla.mozilla.org/");
+            var summarystring = $scope.modalSummary;
+            var productString = "";
+            var componentString = "";
+            var isProductSelected = false;
+
+            $(':input','#modalForm').attr("disabled",true);
+
+            var productRadios = $("#modalForm input[name='productGroup'");
+            if(productRadios && productRadios.length) {
+                for(var i=0, len=productRadios.length; i<len; i++) {
+                    if(productRadios[i].checked) {
+                        productString += productRadios[i].value.split(" :: ")[0];
+                        componentString += productRadios[i].value.split(" :: ")[1];
+                        isProductSelected = true;
+                        break;
+                    }
+                }
+            } else {
+                if(productRadios && productRadios.checked) {
+                    productString += productRadios.value.split(" :: ")[0];
+                    componentString += productRadios.value.split(" :: ")[1];
+                    isProductSelected = true;
+                }
+            }
+
+            if(!isProductSelected) {
+                alert("Please select (or search and select) a product/component pair to continue");
+                return;
+            }
+            var logstrings = "";
+            var logcheckboxes = document.getElementById("modalLogLinkCheckboxes").getElementsByTagName("input");
+
+            for(var i=0;i<logcheckboxes.length;i++) {
+                if(logcheckboxes[i].checked) {
+                    logstrings += logcheckboxes[i].nextElementSibling.href + "\n\n";
+                }
+            }
+
+            // Fetch product information from bugzilla to get version numbers, then submit the new bug
+            // Only request the versions because some products take quite a long time to fetch the full object
+            $.ajax(bugzillaRoot + "rest/product/" + productString + "?include_fields=versions").done(function(productJSON) {
+                var productObject = productJSON.products[0];
+                $http({
+                    //url: bugzillaRoot + "rest/bug?api_key=qF8lX6AyGjcZcmSV4tZTmy2F2PbBycQdB9lsp8cB",
+                    url: "api/bugzilla/create_bug/",
+                    method: "POST",
+                    data: {
+                        "product": productString,
+                        "component": componentString,
+                        "summary": summarystring,
+                        "keywords": "intermittent-failure",
+                        // XXX This takes the last version returned from the product query, probably should be smarter about this in the future...
+                        "version": productObject.versions[productObject.versions.length-1].name,
+                        "description": logstrings + document.getElementById("modalComment").value,
```

Comment:
I'm sure you're getting the feel for this stuff by now, but instead of getting the ``modalComment`` this way, you can just add ``ng-model="modalComment"`` in the HTML for the textarea, and then here reference it as ``$scope.modalComment``